src/multiversioning.zig

```diff
@@ -1054,7 +1054,13 @@ pub const Multiversion = struct {
     }
 
     pub fn exec_latest(self: *Multiversion) !noreturn {
-        assert(self.stage == .ready);
+        // target_fd is only modified in target_update() which happens synchronously.
+        assert(self.stage != .target_update);
```

Comment:
Do we want `!= .err`?

Reply:
Ah, ok, no, we _don't_ need `.err` --- even if we failed to update our `target_fd`, the old version is still fine!

Let's though pass in the version we are intended to switch to as an argument, so that we can check, at the last moment, that `target_fd` indeed contains the version we want to execute. 

We have that check in main as a condition:

```
    for (replica.releases_bundled.const_slice()) |release_bundled| {
        if (release_bundled.value == release.value) break;
    } else {
        log.err("{}: release_execute: release {} is not available;" ++
            "upgrade (or downgrade) the binary", .{
            replica.replica,
            release,
        });
        @panic("release not available");
    }
```

But we should also do an equivalenet at the latest possible moment as an assert.

Reply:
> Ah, ok, no, we don't need .err --- even if we failed to update our target_fd, the old version is still fine!

Yep!

> Let's though pass in the version we are intended to switch to as an argument, so that we can check, at the last moment, that target_fd indeed contains the version we want to execute.

Good idea, will do.